File: app/utils/token.py
import string
import random
import jwt

from app.config import settings


def get_password(size: int):
    chars = string.ascii_uppercase + string.digits
    return ''.join(random.choice(chars) for _ in range(size))


def decode_jwt(token: str, check_expiration: bool = True):
    payload = jwt.decode(token, algorithms='HS256', key=settings.SECRET_KEY, verify_expiration=check_expiration)
    return payload


def encode_jwt(payload: dict):
    token = jwt.encode(payload, key=settings.SECRET_KEY)
    return token.decode('utf-8')


def get_payload(token: str, check_expiration: bool = True, raise_error: bool = True):
    try:
        payload = decode_jwt(token, check_expiration=check_expiration)
    except jwt.exceptions.PyJWTError as e:
        if raise_error:
            raise e

        return None
    return payload

File: auth/app/utils/token.js
const jwt = require('jsonwebtoken');

const AUTH_TOKEN_EXPIRATION = 3 * 60 * 60

const SECRET_KEY = 'backendengineer'

function generate_access_token(phone, name, password, register_at, role) {
    const expiration = Date.now() + AUTH_TOKEN_EXPIRATION
    const payload = {
        'phone': phone,
        'name': name,
        'password': password,
        'register_at': register_at,
        'role': role,
        'exp': expiration.getTime()
    }
    return encode_jwt(payload)
}

function get_password(size) {
    let result = '';
    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890';
    const charactersLength = characters.length;
    for (let i = 0; i < size; i++) {
        result += characters.charAt(Math.floor(Math.random() * charactersLength));
    }
    return result;
}

function decode_jwt(token, check_expiration = true) {
    return jwt.verify(token, SECRET_KEY, {algorithm: 'HS256'}, check_expiration);
}


function encode_jwt(payload) {
    return jwt.sign(payload, SECRET_KEY, {algorithm: 'HS256'})
}

function get_payload(token, check_expiration = true, raise_error = true) {
    try {
        return jwt.verify(token, SECRET_KEY, {algorithm: 'HS256'})
    } catch (err) {
        if (raise_error) {
            throw err
        }
        return null
    }
}


module.exports = {
    generate_access_token,
    get_password,
    decode_jwt,
    encode_jwt,
    get_payload
}